src/service/AppealStorageService.ts

```diff
@@ -0,0 +1,34 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+
+import { Appeal } from 'app/models/Appeal';
+
+export class AppealStorageService {
+
+    constructor(private readonly uri: string) {
+        this.uri = uri;
+    }
+
+    public async save(appealData: Appeal, token: string): Promise<any> {
+
+        if (token == null) {
+            throw new Error('Token is missing');
+        }
+
+        const config: AxiosRequestConfig = {
+            headers: {
+                Accept: 'application/json',
+                Authorization: 'Bearer ' + token
+            }
+        };
+
+        const uri: string = `${this.uri}/companies/${appealData.penaltyIdentifier.companyNumber}/appeals`;
+
+        console.log('Making a POST request to ' + uri);
+
+        return await axios
+            .post(uri, appealData, config)
+            .then((response: AxiosResponse) => {
+                return response;
```

Comment:
I am not a big fan of this. Returning whole response unnecessarily leaks HTTP context outside of the service. Service in its purpose should provide encapsulation so that consumers does not have to think in context of HTTP requests. This change makes design rubbish.

Please revert and make service either return resource ID or error which is not HTTP specific.

Reply:
we are now returning a resource id (location header)